packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -511,4 +512,13 @@ export class CommonFrontendContribution implements MenuContribution, CommandCont
     protected async openAbout() {
         this.aboutDialog.open();
     }
+
+    onWillStop(event: OnWillStopEvent) {
```

Comment:
How about:
```ts
    onWillStop(): boolean {
        if (this.shell.canSaveAll()) {
            setTimeout(() => this.messageService.warn('Some documents should be saved, data will be lost otherwise.'));
            return true;
        }
        return false;
    }
```